src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx

```diff
@@ -19,46 +42,78 @@ export default class SentryAppDetailsModal extends React.Component {
     closeModal: PropTypes.func.isRequired,
   };
 
-  render() {
+  getEndpoints() {
+    const {sentryApp} = this.props;
+    return [['featureData', `/sentry-apps/${sentryApp.slug}/features/`]];
+  }
+
+  featureTags(features) {
+    return features.map(feature => {
+      const feat = feature.featureGate.replace(/integrations/g, '');
+      return <StyledTag key={feat}>{feat.replace(/-/g, ' ')}</StyledTag>;
+    });
+  }
+
+  renderBody() {
     const {sentryApp, closeModal, onInstall, isInstalled, organization} = this.props;
+    const {featureData} = this.state;
+    // Prepare the features list
+    const features = (featureData || []).map(f => ({
+      featureGate: f.featureGate,
+      description: (
+        <span dangerouslySetInnerHTML={{__html: singleLineRenderer(f.description)}} />
+      ),
+    }));
+
+    const featureListHooks = HookStore.get('integrations:feature-gates');
+    featureListHooks.push(() => defaultFeatureGateComponents);
 
+    const {FeatureList, IntegrationFeatures} = featureListHooks[0]();
```

Comment:
So the `defaultFeatureGateComponent` list for if you are just running sentry to make this compatible for on-prem people. 

> Is the push in case it returns an empty list?

This is so we use the pretty markdown and plan specific information for sentry.io

I forget exactly know how the `HookStore` works so maybe @EvanPurkhiser can shed light on the 'returns list of functions' question.

Reply:
@MeredithAnya I agree with @mnoble that the code makes this a bit confusing to understand what's going on, though I see this is used other places in the code base. I'm wondering if there is a clearer way to get the same behavior. Here is my suggestion:
```
const featureHook = HookStore.get('integrations:feature-gates')[0] || defaultFeatureGateComponents;
const {FeatureList, IntegrationFeatures} = featureHook();
```

This also avoids mutating the array that is stored in the `HookStore` with key `integrations:feature-gates`. The mutation probably does not have any consequences in this case but in general, it's better to avoid mutation unless that is the explicit goal.

Reply:
This was my doing.

Yes we push it onto the end in the case that it's empty.

```jsx
const featureHook = HookStore.get('integrations:feature-gates')[0] || defaultFeatureGateComponents;
```

This would need to be written as 

```jsx
const featureHook = HookStore.get('integrations:feature-gates')[0] || (() => defaultFeatureGateComponents);
```

Since we call the hookstore function to get our components back.

Reply:
> I'm not sure what the above code is doing. It looks like HookStore returns a list of functions

This is historically just the convention that we have typically used for hookstore. I think it's not great considering there are actually a lot of times where we do *not* want a list. In fact, if you look around the code base, I would say a majority of the times we just pop off the first function and call it.

Reply:
@EvanPurkhiser Does it make sense to make a `getFirst` method on the HookStore? Probably outside the scope of this PR but if what you are saying is true then getting the full array is just a hassle.

Reply:
It would probably make more sense to re-evaluate how we're using hookstore. Not sure how much you've seen it yet, but it's basically a bridge between sentry and getsentry, a way to add frontend features specific to the SASS product